src/components/Card.js

```diff
@@ -1,25 +1,43 @@
 //
 
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteBtnClick,
+    handleLikeBtnClick
+  ) {
     this.name = name;
     this.link = link;
+    this.id = _id;
+    this._initialState = isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteBtnClick = handleDeleteBtnClick;
+    this._handleLikeBtnClick = handleLikeBtnClick;
   }
 
-  _handleLikeBtnClick = () => {
+  setLikeBtnState = () => {
     this._likeBtn.classList.toggle("card__like-button_active");
   };
 
-  _handleDeleteBtnClick = () => {
+  isLiked = () => {
+    return this._likeBtn.classList.contains("card__like-button_active");
+  };
+
+  handleDelete = () => {
     this._cardElement.remove();
```

Comment:
**Could be improved**

The best thing to do after deleting a card is to **remove the link** to the DOM element: `this._cardElement = null;`. It helps javascript garbage collector.

More info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_Management